// @aws-sdk/client-sfn/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin, getLongPollPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultSFNHttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { CreateActivity$, CreateStateMachineAlias$, CreateStateMachine$, DeleteActivity$, DeleteStateMachineAlias$, DeleteStateMachine$, DeleteStateMachineVersion$, DescribeActivity$, DescribeExecution$, DescribeMapRun$, DescribeStateMachineAlias$, DescribeStateMachine$, DescribeStateMachineForExecution$, GetActivityTask$, GetExecutionHistory$, ListActivities$, ListExecutions$, ListMapRuns$, ListStateMachineAliases$, ListStateMachines$, ListStateMachineVersions$, ListTagsForResource$, PublishStateMachineVersion$, RedriveExecution$, SendTaskFailure$, SendTaskHeartbeat$, SendTaskSuccess$, StartExecution$, StartSyncExecution$, StopExecution$, TagResource$, TestState$, UntagResource$, UpdateMapRun$, UpdateStateMachineAlias$, UpdateStateMachine$, ValidateStateMachineDefinition$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
__exportStar(require("./models/errors"), exports);
const { SFNServiceException } = require("./models/SFNServiceException");
exports.SFNServiceException = SFNServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "states",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class SFNClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultSFNHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class CreateActivityCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "CreateActivity", {})
    .n("SFNClient", "CreateActivityCommand")
    .sc(CreateActivity$)
    .build() {
}

class CreateStateMachineAliasCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "CreateStateMachineAlias", {})
    .n("SFNClient", "CreateStateMachineAliasCommand")
    .sc(CreateStateMachineAlias$)
    .build() {
}

class CreateStateMachineCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "CreateStateMachine", {})
    .n("SFNClient", "CreateStateMachineCommand")
    .sc(CreateStateMachine$)
    .build() {
}

class DeleteActivityCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DeleteActivity", {})
    .n("SFNClient", "DeleteActivityCommand")
    .sc(DeleteActivity$)
    .build() {
}

class DeleteStateMachineAliasCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DeleteStateMachineAlias", {})
    .n("SFNClient", "DeleteStateMachineAliasCommand")
    .sc(DeleteStateMachineAlias$)
    .build() {
}

class DeleteStateMachineCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DeleteStateMachine", {})
    .n("SFNClient", "DeleteStateMachineCommand")
    .sc(DeleteStateMachine$)
    .build() {
}

class DeleteStateMachineVersionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DeleteStateMachineVersion", {})
    .n("SFNClient", "DeleteStateMachineVersionCommand")
    .sc(DeleteStateMachineVersion$)
    .build() {
}

class DescribeActivityCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DescribeActivity", {})
    .n("SFNClient", "DescribeActivityCommand")
    .sc(DescribeActivity$)
    .build() {
}

class DescribeExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DescribeExecution", {})
    .n("SFNClient", "DescribeExecutionCommand")
    .sc(DescribeExecution$)
    .build() {
}

class DescribeMapRunCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DescribeMapRun", {})
    .n("SFNClient", "DescribeMapRunCommand")
    .sc(DescribeMapRun$)
    .build() {
}

class DescribeStateMachineAliasCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DescribeStateMachineAlias", {})
    .n("SFNClient", "DescribeStateMachineAliasCommand")
    .sc(DescribeStateMachineAlias$)
    .build() {
}

class DescribeStateMachineCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DescribeStateMachine", {})
    .n("SFNClient", "DescribeStateMachineCommand")
    .sc(DescribeStateMachine$)
    .build() {
}

class DescribeStateMachineForExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "DescribeStateMachineForExecution", {})
    .n("SFNClient", "DescribeStateMachineForExecutionCommand")
    .sc(DescribeStateMachineForExecution$)
    .build() {
}

class GetActivityTaskCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        getLongPollPlugin(config),
    ];
})
    .s("AWSStepFunctions", "GetActivityTask", {})
    .n("SFNClient", "GetActivityTaskCommand")
    .sc(GetActivityTask$)
    .build() {
}

class GetExecutionHistoryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "GetExecutionHistory", {})
    .n("SFNClient", "GetExecutionHistoryCommand")
    .sc(GetExecutionHistory$)
    .build() {
}

class ListActivitiesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListActivities", {})
    .n("SFNClient", "ListActivitiesCommand")
    .sc(ListActivities$)
    .build() {
}

class ListExecutionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListExecutions", {})
    .n("SFNClient", "ListExecutionsCommand")
    .sc(ListExecutions$)
    .build() {
}

class ListMapRunsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListMapRuns", {})
    .n("SFNClient", "ListMapRunsCommand")
    .sc(ListMapRuns$)
    .build() {
}

class ListStateMachineAliasesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListStateMachineAliases", {})
    .n("SFNClient", "ListStateMachineAliasesCommand")
    .sc(ListStateMachineAliases$)
    .build() {
}

class ListStateMachinesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListStateMachines", {})
    .n("SFNClient", "ListStateMachinesCommand")
    .sc(ListStateMachines$)
    .build() {
}

class ListStateMachineVersionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListStateMachineVersions", {})
    .n("SFNClient", "ListStateMachineVersionsCommand")
    .sc(ListStateMachineVersions$)
    .build() {
}

class ListTagsForResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ListTagsForResource", {})
    .n("SFNClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class PublishStateMachineVersionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "PublishStateMachineVersion", {})
    .n("SFNClient", "PublishStateMachineVersionCommand")
    .sc(PublishStateMachineVersion$)
    .build() {
}

class RedriveExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "RedriveExecution", {})
    .n("SFNClient", "RedriveExecutionCommand")
    .sc(RedriveExecution$)
    .build() {
}

class SendTaskFailureCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "SendTaskFailure", {})
    .n("SFNClient", "SendTaskFailureCommand")
    .sc(SendTaskFailure$)
    .build() {
}

class SendTaskHeartbeatCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "SendTaskHeartbeat", {})
    .n("SFNClient", "SendTaskHeartbeatCommand")
    .sc(SendTaskHeartbeat$)
    .build() {
}

class SendTaskSuccessCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "SendTaskSuccess", {})
    .n("SFNClient", "SendTaskSuccessCommand")
    .sc(SendTaskSuccess$)
    .build() {
}

class StartExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "StartExecution", {})
    .n("SFNClient", "StartExecutionCommand")
    .sc(StartExecution$)
    .build() {
}

class StartSyncExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "StartSyncExecution", {})
    .n("SFNClient", "StartSyncExecutionCommand")
    .sc(StartSyncExecution$)
    .build() {
}

class StopExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "StopExecution", {})
    .n("SFNClient", "StopExecutionCommand")
    .sc(StopExecution$)
    .build() {
}

class TagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "TagResource", {})
    .n("SFNClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class TestStateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "TestState", {})
    .n("SFNClient", "TestStateCommand")
    .sc(TestState$)
    .build() {
}

class UntagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "UntagResource", {})
    .n("SFNClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateMapRunCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "UpdateMapRun", {})
    .n("SFNClient", "UpdateMapRunCommand")
    .sc(UpdateMapRun$)
    .build() {
}

class UpdateStateMachineAliasCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "UpdateStateMachineAlias", {})
    .n("SFNClient", "UpdateStateMachineAliasCommand")
    .sc(UpdateStateMachineAlias$)
    .build() {
}

class UpdateStateMachineCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "UpdateStateMachine", {})
    .n("SFNClient", "UpdateStateMachineCommand")
    .sc(UpdateStateMachine$)
    .build() {
}

class ValidateStateMachineDefinitionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSStepFunctions", "ValidateStateMachineDefinition", {})
    .n("SFNClient", "ValidateStateMachineDefinitionCommand")
    .sc(ValidateStateMachineDefinition$)
    .build() {
}

const paginateGetExecutionHistory = createPaginator(SFNClient, GetExecutionHistoryCommand, "nextToken", "nextToken", "maxResults");

const paginateListActivities = createPaginator(SFNClient, ListActivitiesCommand, "nextToken", "nextToken", "maxResults");

const paginateListExecutions = createPaginator(SFNClient, ListExecutionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListMapRuns = createPaginator(SFNClient, ListMapRunsCommand, "nextToken", "nextToken", "maxResults");

const paginateListStateMachines = createPaginator(SFNClient, ListStateMachinesCommand, "nextToken", "nextToken", "maxResults");

const commands = {
    CreateActivityCommand,
    CreateStateMachineCommand,
    CreateStateMachineAliasCommand,
    DeleteActivityCommand,
    DeleteStateMachineCommand,
    DeleteStateMachineAliasCommand,
    DeleteStateMachineVersionCommand,
    DescribeActivityCommand,
    DescribeExecutionCommand,
    DescribeMapRunCommand,
    DescribeStateMachineCommand,
    DescribeStateMachineAliasCommand,
    DescribeStateMachineForExecutionCommand,
    GetActivityTaskCommand,
    GetExecutionHistoryCommand,
    ListActivitiesCommand,
    ListExecutionsCommand,
    ListMapRunsCommand,
    ListStateMachineAliasesCommand,
    ListStateMachinesCommand,
    ListStateMachineVersionsCommand,
    ListTagsForResourceCommand,
    PublishStateMachineVersionCommand,
    RedriveExecutionCommand,
    SendTaskFailureCommand,
    SendTaskHeartbeatCommand,
    SendTaskSuccessCommand,
    StartExecutionCommand,
    StartSyncExecutionCommand,
    StopExecutionCommand,
    TagResourceCommand,
    TestStateCommand,
    UntagResourceCommand,
    UpdateMapRunCommand,
    UpdateStateMachineCommand,
    UpdateStateMachineAliasCommand,
    ValidateStateMachineDefinitionCommand,
};
const paginators = {
    paginateGetExecutionHistory,
    paginateListActivities,
    paginateListExecutions,
    paginateListMapRuns,
    paginateListStateMachines,
};
class SFN extends SFNClient {
}
createAggregatedClient(commands, SFN, { paginators });

const EncryptionType = {
    AWS_OWNED_KEY: "AWS_OWNED_KEY",
    CUSTOMER_MANAGED_KMS_KEY: "CUSTOMER_MANAGED_KMS_KEY",
};
const LogLevel = {
    ALL: "ALL",
    ERROR: "ERROR",
    FATAL: "FATAL",
    OFF: "OFF",
};
const StateMachineType = {
    EXPRESS: "EXPRESS",
    STANDARD: "STANDARD",
};
const ValidationExceptionReason = {
    API_DOES_NOT_SUPPORT_LABELED_ARNS: "API_DOES_NOT_SUPPORT_LABELED_ARNS",
    CANNOT_UPDATE_COMPLETED_MAP_RUN: "CANNOT_UPDATE_COMPLETED_MAP_RUN",
    INVALID_ROUTING_CONFIGURATION: "INVALID_ROUTING_CONFIGURATION",
    MISSING_REQUIRED_PARAMETER: "MISSING_REQUIRED_PARAMETER",
};
const IncludedData = {
    ALL_DATA: "ALL_DATA",
    METADATA_ONLY: "METADATA_ONLY",
};
const ExecutionRedriveStatus = {
    NOT_REDRIVABLE: "NOT_REDRIVABLE",
    REDRIVABLE: "REDRIVABLE",
    REDRIVABLE_BY_MAP_RUN: "REDRIVABLE_BY_MAP_RUN",
};
const ExecutionStatus = {
    ABORTED: "ABORTED",
    FAILED: "FAILED",
    PENDING_REDRIVE: "PENDING_REDRIVE",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
    TIMED_OUT: "TIMED_OUT",
};
const KmsKeyState = {
    CREATING: "CREATING",
    DISABLED: "DISABLED",
    PENDING_DELETION: "PENDING_DELETION",
    PENDING_IMPORT: "PENDING_IMPORT",
    UNAVAILABLE: "UNAVAILABLE",
};
const MapRunStatus = {
    ABORTED: "ABORTED",
    FAILED: "FAILED",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
};
const StateMachineStatus = {
    ACTIVE: "ACTIVE",
    DELETING: "DELETING",
};
const HistoryEventType = {
    ActivityFailed: "ActivityFailed",
    ActivityScheduleFailed: "ActivityScheduleFailed",
    ActivityScheduled: "ActivityScheduled",
    ActivityStarted: "ActivityStarted",
    ActivitySucceeded: "ActivitySucceeded",
    ActivityTimedOut: "ActivityTimedOut",
    ChoiceStateEntered: "ChoiceStateEntered",
    ChoiceStateExited: "ChoiceStateExited",
    EvaluationFailed: "EvaluationFailed",
    ExecutionAborted: "ExecutionAborted",
    ExecutionFailed: "ExecutionFailed",
    ExecutionRedriven: "ExecutionRedriven",
    ExecutionStarted: "ExecutionStarted",
    ExecutionSucceeded: "ExecutionSucceeded",
    ExecutionTimedOut: "ExecutionTimedOut",
    FailStateEntered: "FailStateEntered",
    LambdaFunctionFailed: "LambdaFunctionFailed",
    LambdaFunctionScheduleFailed: "LambdaFunctionScheduleFailed",
    LambdaFunctionScheduled: "LambdaFunctionScheduled",
    LambdaFunctionStartFailed: "LambdaFunctionStartFailed",
    LambdaFunctionStarted: "LambdaFunctionStarted",
    LambdaFunctionSucceeded: "LambdaFunctionSucceeded",
    LambdaFunctionTimedOut: "LambdaFunctionTimedOut",
    MapIterationAborted: "MapIterationAborted",
    MapIterationFailed: "MapIterationFailed",
    MapIterationStarted: "MapIterationStarted",
    MapIterationSucceeded: "MapIterationSucceeded",
    MapRunAborted: "MapRunAborted",
    MapRunFailed: "MapRunFailed",
    MapRunRedriven: "MapRunRedriven",
    MapRunStarted: "MapRunStarted",
    MapRunSucceeded: "MapRunSucceeded",
    MapStateAborted: "MapStateAborted",
    MapStateEntered: "MapStateEntered",
    MapStateExited: "MapStateExited",
    MapStateFailed: "MapStateFailed",
    MapStateStarted: "MapStateStarted",
    MapStateSucceeded: "MapStateSucceeded",
    ParallelStateAborted: "ParallelStateAborted",
    ParallelStateEntered: "ParallelStateEntered",
    ParallelStateExited: "ParallelStateExited",
    ParallelStateFailed: "ParallelStateFailed",
    ParallelStateStarted: "ParallelStateStarted",
    ParallelStateSucceeded: "ParallelStateSucceeded",
    PassStateEntered: "PassStateEntered",
    PassStateExited: "PassStateExited",
    SucceedStateEntered: "SucceedStateEntered",
    SucceedStateExited: "SucceedStateExited",
    TaskFailed: "TaskFailed",
    TaskScheduled: "TaskScheduled",
    TaskStartFailed: "TaskStartFailed",
    TaskStarted: "TaskStarted",
    TaskStateAborted: "TaskStateAborted",
    TaskStateEntered: "TaskStateEntered",
    TaskStateExited: "TaskStateExited",
    TaskSubmitFailed: "TaskSubmitFailed",
    TaskSubmitted: "TaskSubmitted",
    TaskSucceeded: "TaskSucceeded",
    TaskTimedOut: "TaskTimedOut",
    WaitStateAborted: "WaitStateAborted",
    WaitStateEntered: "WaitStateEntered",
    WaitStateExited: "WaitStateExited",
};
const ExecutionRedriveFilter = {
    NOT_REDRIVEN: "NOT_REDRIVEN",
    REDRIVEN: "REDRIVEN",
};
const SyncExecutionStatus = {
    FAILED: "FAILED",
    SUCCEEDED: "SUCCEEDED",
    TIMED_OUT: "TIMED_OUT",
};
const InspectionLevel = {
    DEBUG: "DEBUG",
    INFO: "INFO",
    TRACE: "TRACE",
};
const MockResponseValidationMode = {
    NONE: "NONE",
    PRESENT: "PRESENT",
    STRICT: "STRICT",
};
const TestExecutionStatus = {
    CAUGHT_ERROR: "CAUGHT_ERROR",
    FAILED: "FAILED",
    RETRIABLE: "RETRIABLE",
    SUCCEEDED: "SUCCEEDED",
};
const ValidateStateMachineDefinitionSeverity = {
    ERROR: "ERROR",
    WARNING: "WARNING",
};
const ValidateStateMachineDefinitionResultCode = {
    FAIL: "FAIL",
    OK: "OK",
};

exports.CreateActivityCommand = CreateActivityCommand;
exports.CreateStateMachineAliasCommand = CreateStateMachineAliasCommand;
exports.CreateStateMachineCommand = CreateStateMachineCommand;
exports.DeleteActivityCommand = DeleteActivityCommand;
exports.DeleteStateMachineAliasCommand = DeleteStateMachineAliasCommand;
exports.DeleteStateMachineCommand = DeleteStateMachineCommand;
exports.DeleteStateMachineVersionCommand = DeleteStateMachineVersionCommand;
exports.DescribeActivityCommand = DescribeActivityCommand;
exports.DescribeExecutionCommand = DescribeExecutionCommand;
exports.DescribeMapRunCommand = DescribeMapRunCommand;
exports.DescribeStateMachineAliasCommand = DescribeStateMachineAliasCommand;
exports.DescribeStateMachineCommand = DescribeStateMachineCommand;
exports.DescribeStateMachineForExecutionCommand = DescribeStateMachineForExecutionCommand;
exports.EncryptionType = EncryptionType;
exports.ExecutionRedriveFilter = ExecutionRedriveFilter;
exports.ExecutionRedriveStatus = ExecutionRedriveStatus;
exports.ExecutionStatus = ExecutionStatus;
exports.GetActivityTaskCommand = GetActivityTaskCommand;
exports.GetExecutionHistoryCommand = GetExecutionHistoryCommand;
exports.HistoryEventType = HistoryEventType;
exports.IncludedData = IncludedData;
exports.InspectionLevel = InspectionLevel;
exports.KmsKeyState = KmsKeyState;
exports.ListActivitiesCommand = ListActivitiesCommand;
exports.ListExecutionsCommand = ListExecutionsCommand;
exports.ListMapRunsCommand = ListMapRunsCommand;
exports.ListStateMachineAliasesCommand = ListStateMachineAliasesCommand;
exports.ListStateMachineVersionsCommand = ListStateMachineVersionsCommand;
exports.ListStateMachinesCommand = ListStateMachinesCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.LogLevel = LogLevel;
exports.MapRunStatus = MapRunStatus;
exports.MockResponseValidationMode = MockResponseValidationMode;
exports.PublishStateMachineVersionCommand = PublishStateMachineVersionCommand;
exports.RedriveExecutionCommand = RedriveExecutionCommand;
exports.SFN = SFN;
exports.SFNClient = SFNClient;
exports.SendTaskFailureCommand = SendTaskFailureCommand;
exports.SendTaskHeartbeatCommand = SendTaskHeartbeatCommand;
exports.SendTaskSuccessCommand = SendTaskSuccessCommand;
exports.StartExecutionCommand = StartExecutionCommand;
exports.StartSyncExecutionCommand = StartSyncExecutionCommand;
exports.StateMachineStatus = StateMachineStatus;
exports.StateMachineType = StateMachineType;
exports.StopExecutionCommand = StopExecutionCommand;
exports.SyncExecutionStatus = SyncExecutionStatus;
exports.TagResourceCommand = TagResourceCommand;
exports.TestExecutionStatus = TestExecutionStatus;
exports.TestStateCommand = TestStateCommand;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateMapRunCommand = UpdateMapRunCommand;
exports.UpdateStateMachineAliasCommand = UpdateStateMachineAliasCommand;
exports.UpdateStateMachineCommand = UpdateStateMachineCommand;
exports.ValidateStateMachineDefinitionCommand = ValidateStateMachineDefinitionCommand;
exports.ValidateStateMachineDefinitionResultCode = ValidateStateMachineDefinitionResultCode;
exports.ValidateStateMachineDefinitionSeverity = ValidateStateMachineDefinitionSeverity;
exports.ValidationExceptionReason = ValidationExceptionReason;
exports.paginateGetExecutionHistory = paginateGetExecutionHistory;
exports.paginateListActivities = paginateListActivities;
exports.paginateListExecutions = paginateListExecutions;
exports.paginateListMapRuns = paginateListMapRuns;
exports.paginateListStateMachines = paginateListStateMachines;
