// @aws-sdk/client-sfn/dist-cjs/runtimeConfig.shared.js
const { AwsSdkSigV4Signer } = require("@aws-sdk/core/httpAuthSchemes");
const { AwsJson1_0Protocol } = require("@aws-sdk/core/protocols");
const { NoOpLogger } = require("@smithy/core/client");
const { parseUrl } = require("@smithy/core/protocols");
const { fromBase64, fromUtf8, toBase64, toUtf8 } = require("@smithy/core/serde");
const { defaultSFNHttpAuthSchemeProvider } = require("./auth/httpAuthSchemeProvider");
const { defaultEndpointResolver } = require("./endpoint/endpointResolver");
const { errorTypeRegistries } = require("./schemas/schemas_0");
exports.getRuntimeConfig = (config) => {
    return {
        apiVersion: "2016-11-23",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultSFNHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsJson1_0Protocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.sfn",
            errorTypeRegistries,
            xmlNamespace: "http://swf.amazonaws.com/doc/2015-07-20/",
            version: "2016-11-23",
            serviceTarget: "AWSStepFunctions",
        },
        serviceId: config?.serviceId ?? "SFN",
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};
