// @aws-sdk/client-sfn/dist-cjs/models/errors.js
const { SFNServiceException: __BaseException } = require("./SFNServiceException");
exports.ActivityAlreadyExists = class ActivityAlreadyExists extends __BaseException {
    name = "ActivityAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ActivityAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ActivityAlreadyExists.prototype);
    }
};
exports.ActivityDoesNotExist = class ActivityDoesNotExist extends __BaseException {
    name = "ActivityDoesNotExist";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ActivityDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ActivityDoesNotExist.prototype);
    }
};
exports.ActivityLimitExceeded = class ActivityLimitExceeded extends __BaseException {
    name = "ActivityLimitExceeded";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ActivityLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ActivityLimitExceeded.prototype);
    }
};
exports.ActivityWorkerLimitExceeded = class ActivityWorkerLimitExceeded extends __BaseException {
    name = "ActivityWorkerLimitExceeded";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ActivityWorkerLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ActivityWorkerLimitExceeded.prototype);
    }
};
exports.InvalidEncryptionConfiguration = class InvalidEncryptionConfiguration extends __BaseException {
    name = "InvalidEncryptionConfiguration";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidEncryptionConfiguration",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidEncryptionConfiguration.prototype);
    }
};
exports.InvalidName = class InvalidName extends __BaseException {
    name = "InvalidName";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidName",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidName.prototype);
    }
};
exports.KmsAccessDeniedException = class KmsAccessDeniedException extends __BaseException {
    name = "KmsAccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsAccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsAccessDeniedException.prototype);
    }
};
exports.KmsThrottlingException = class KmsThrottlingException extends __BaseException {
    name = "KmsThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsThrottlingException.prototype);
    }
};
exports.TooManyTags = class TooManyTags extends __BaseException {
    name = "TooManyTags";
    $fault = "client";
    resourceName;
    constructor(opts) {
        super({
            name: "TooManyTags",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTags.prototype);
        this.resourceName = opts.resourceName;
    }
};
exports.ConflictException = class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
};
exports.InvalidArn = class InvalidArn extends __BaseException {
    name = "InvalidArn";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidArn",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidArn.prototype);
    }
};
exports.InvalidDefinition = class InvalidDefinition extends __BaseException {
    name = "InvalidDefinition";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDefinition",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDefinition.prototype);
    }
};
exports.InvalidLoggingConfiguration = class InvalidLoggingConfiguration extends __BaseException {
    name = "InvalidLoggingConfiguration";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidLoggingConfiguration",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLoggingConfiguration.prototype);
    }
};
exports.InvalidTracingConfiguration = class InvalidTracingConfiguration extends __BaseException {
    name = "InvalidTracingConfiguration";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTracingConfiguration",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTracingConfiguration.prototype);
    }
};
exports.StateMachineAlreadyExists = class StateMachineAlreadyExists extends __BaseException {
    name = "StateMachineAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "StateMachineAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StateMachineAlreadyExists.prototype);
    }
};
exports.StateMachineDeleting = class StateMachineDeleting extends __BaseException {
    name = "StateMachineDeleting";
    $fault = "client";
    constructor(opts) {
        super({
            name: "StateMachineDeleting",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StateMachineDeleting.prototype);
    }
};
exports.StateMachineLimitExceeded = class StateMachineLimitExceeded extends __BaseException {
    name = "StateMachineLimitExceeded";
    $fault = "client";
    constructor(opts) {
        super({
            name: "StateMachineLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StateMachineLimitExceeded.prototype);
    }
};
exports.StateMachineTypeNotSupported = class StateMachineTypeNotSupported extends __BaseException {
    name = "StateMachineTypeNotSupported";
    $fault = "client";
    constructor(opts) {
        super({
            name: "StateMachineTypeNotSupported",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StateMachineTypeNotSupported.prototype);
    }
};
exports.ValidationException = class ValidationException extends __BaseException {
    name = "ValidationException";
    $fault = "client";
    reason;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.reason = opts.reason;
    }
};
exports.ResourceNotFound = class ResourceNotFound extends __BaseException {
    name = "ResourceNotFound";
    $fault = "client";
    resourceName;
    constructor(opts) {
        super({
            name: "ResourceNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFound.prototype);
        this.resourceName = opts.resourceName;
    }
};
exports.ServiceQuotaExceededException = class ServiceQuotaExceededException extends __BaseException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
};
exports.ExecutionDoesNotExist = class ExecutionDoesNotExist extends __BaseException {
    name = "ExecutionDoesNotExist";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ExecutionDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ExecutionDoesNotExist.prototype);
    }
};
exports.KmsInvalidStateException = class KmsInvalidStateException extends __BaseException {
    name = "KmsInvalidStateException";
    $fault = "client";
    kmsKeyState;
    constructor(opts) {
        super({
            name: "KmsInvalidStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsInvalidStateException.prototype);
        this.kmsKeyState = opts.kmsKeyState;
    }
};
exports.StateMachineDoesNotExist = class StateMachineDoesNotExist extends __BaseException {
    name = "StateMachineDoesNotExist";
    $fault = "client";
    constructor(opts) {
        super({
            name: "StateMachineDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, StateMachineDoesNotExist.prototype);
    }
};
exports.InvalidToken = class InvalidToken extends __BaseException {
    name = "InvalidToken";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidToken",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidToken.prototype);
    }
};
exports.ExecutionLimitExceeded = class ExecutionLimitExceeded extends __BaseException {
    name = "ExecutionLimitExceeded";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ExecutionLimitExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ExecutionLimitExceeded.prototype);
    }
};
exports.ExecutionNotRedrivable = class ExecutionNotRedrivable extends __BaseException {
    name = "ExecutionNotRedrivable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ExecutionNotRedrivable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ExecutionNotRedrivable.prototype);
    }
};
exports.TaskDoesNotExist = class TaskDoesNotExist extends __BaseException {
    name = "TaskDoesNotExist";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TaskDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TaskDoesNotExist.prototype);
    }
};
exports.TaskTimedOut = class TaskTimedOut extends __BaseException {
    name = "TaskTimedOut";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TaskTimedOut",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TaskTimedOut.prototype);
    }
};
exports.InvalidOutput = class InvalidOutput extends __BaseException {
    name = "InvalidOutput";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidOutput",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidOutput.prototype);
    }
};
exports.ExecutionAlreadyExists = class ExecutionAlreadyExists extends __BaseException {
    name = "ExecutionAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ExecutionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ExecutionAlreadyExists.prototype);
    }
};
exports.InvalidExecutionInput = class InvalidExecutionInput extends __BaseException {
    name = "InvalidExecutionInput";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidExecutionInput",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidExecutionInput.prototype);
    }
};
exports.MissingRequiredParameter = class MissingRequiredParameter extends __BaseException {
    name = "MissingRequiredParameter";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MissingRequiredParameter",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MissingRequiredParameter.prototype);
    }
};
